recipes/mysql_master.rb

```diff
@@ -18,49 +18,50 @@
 # limitations under the License.
 #
 
-# set repl password
-::Chef::Recipe.send(:include, Opscode::OpenSSL::Password)
-node.set_unless['mysql-multi']['server_repl_password'] = secure_password
-
+include_recipe 'apt' if node.platform_family?('debian')
 include_recipe 'chef-sugar'
 include_recipe 'mysql-multi::_find_slaves'
-include_recipe 'mysql-multi'
 
-# drop MySQL master specific configuration file
-template '/etc/mysql/conf.d/master.cnf' do
+# set passwords dynamically...
+::Chef::Recipe.send(:include, Opscode::OpenSSL::Password)
+node.set_unless['mysql-multi']['server_root_password'] = secure_password
+
+# creates unique serverid via ipaddress to an int
+require 'ipaddr'
+serverid = IPAddr.new node['ipaddress']
+serverid = serverid.to_i
```

Comment:
It feels like L29-L32 could also be in a helper library or in the mysql_config provider instead of directly in the recipe. Something feels unpleasant about having this require in the recipe. Not sure I've proposed a really great alternative though.